samples/fpgaperf_counter/fpgaperf_counter.c

```diff
@@ -173,7 +173,7 @@ STATIC fpga_result parse_perf_attributes(struct udev_device *dev,
 			globfree(&pglob);
 			return FPGA_EXCEPTION;
 		}
-		if (fscanf(file, "%s", attr_value) != 1) {
+		if (fscanf(file, "%127s", attr_value) != 1) {
```

Comment:
attr_value is 128 bytes?

Reply:
Vulnerable code example
  void main() {
      char s[16];
      scanf("%s",s);
 }

Fixed code example
  void main() {
      char s[16];
      scanf("%15s",s);
 }
In the fixed code, width of the string is provided correctly, ensuring that the destination buffer won't overflow.